controlpanel/frontend/static/components/selectable-rows/selectable-rows.js

```diff
@@ -1,24 +1,92 @@
+/**
+ * Module to allow selecting entire rows in a table instead of using a small checkbox.
+ *
+ * It will disable any in-row actions buttons once a selection is made, but will re-enable them once
+ * all the selections are cleared.
+ *
+ * It can optionally support enabling a (global)
+ * button(s) that lives outside the table.
+ * How to use:
+ *  1. Add the class "selectable-rows" to your <table>
+ *  2. For any button you want to enable when there are selections add the class "selectable-rows__enable-on-selections"
+ *  3. Use the macro `row_selector` to put insert an invisible input box in the first column of your table
+ * The module will automatically disable any action buttons that are in side the row.
+ *
+ * WARNING: This module doesn't support more than 1 table on the same page
+ * @type {{init(): void, onSelectionChanged(*): void, baseClass: string, bindEvents(): void}}
+ */
+
 moj.Modules.selectable_rows = {
-  selector: '.selectable-rows tr',
+  baseClass: "selectable-rows",
 
   init() {
-    if (document.querySelector(this.selector)) {
+    this.tableRowSelector = `.${this.baseClass} tr`;
+    this.tableRowButtonSelector = `${this.tableRowSelector} button`;
+    this.selectionClass = `${this.baseClass}__row-selected`;
+    this.enableOnSelectionSelector = `.${this.baseClass}__enable-on-selections`;
+    this.selectionChangedEventName = `${this.baseClass}:rowSelectionChanged`;
```

Comment:
it doesn't matter and it's merged but maybe we could lose the `row` for brevity and consistency with the `onSelectionChanged` event handler :)